docs/comparison-matrix.md

```diff
@@ -0,0 +1,41 @@
+|--------------------------------------------------------|-------------------|--------------|-----------------|--------------------|----------|--------------|
+| Sum Types                                              | 🟢                | 🟢           | 🔴              | 🟢                 | 🟢       | 🟢           |
+| Record Types                                           | 🟢                | 🟢           | 🟢              | 🟢                 | 🟢       | 🟢           |
+| Product Types                                          | 🔴                | 🔴           | 🔴              | 🟢                 | ❔       | 🔴           |
+| Recursive Types                                        | 🟢                | 🟢           | 🔴              | 🟢                 | 🟢       | ❔           |
+| Type functions (Generics)                              | 🔴                | 🟢           | 🔴              | 🟢                 | 🟢       | 🔴           |
+| Type annotations/constraints                           | 🟢                | 🟢           | 🟢              | 🔵                 | 🟢       | 🟢           |
+| Add new builtin types                                  | 🔴                | 🟢           | 🔴              | 🟢                 | 🔴       | 🔴           |
+| Add new type semantics (e.g. different encodings)      | 🟢                | 🟢           | 🔴              | 🟢                 | 🔴       | 🟢           |
+| Manage type semantics (at language level)              | 🔴                | 🔴           | 🔴              | 🟢                 | 🔴       | 🔴           |
+| Codegen support                                        | 🟢 (Excellent)    | 🟢 (Average) | 🟢 (Excellent)  | 🟡                 | 🟢 (Bad) | 🟢 (Average) |
+| DevOps tooling - build system integration              | 🟢                | 🔴           | ❔              | 🟡                 | 🔴       | 🔴           |
+| Documentation tooling                                  | 🟢                | ❔           | 🟢              | 🔵                 | 🔴       | ❔           |
+| Formatting, linting, and development environment tools | 🟢                | 🔴           | 🟢              | 🔵                 | 🔴       | 🔴           |
+| Documentation tooling                                  | 🟢                | 🔴           | 🟢              | 🔵                 | 🔴       | ❔           |
+| Language checker API                                   | 🟢                | 🔴           | 🟢              | 🟢                 | 🔴       | 🔴           |
+| Codegen API                                            | 🟢                | 🟢           | 🔴              | 🟢                 | 🔴       | 🔴           |
+| Language specification                                 | 🟢                | 🟢           | 🟢              | 🟢                 | 🟢       | 🟢           |
+| Backwards compatibility strategy                       | 🟢                | 🔴           | 🔴              | 🔴                 | 🔴       | 🔴           |
+
```

Comment:
Perhaps adding a chapter that elaborates on what the features are would be nice.